storage/src/test/java/org/apache/kafka/server/log/remote/storage/RemoteLogManagerTest.java

```diff
@@ -3412,8 +3412,14 @@ public void testCopyQuota(boolean quotaExceeded) throws Exception {
             Map<org.apache.kafka.common.MetricName, KafkaMetric> allMetrics = metrics.metrics();
             KafkaMetric avgMetric = allMetrics.get(metrics.metricName("remote-copy-throttle-time-avg", "RemoteLogManager"));
             KafkaMetric maxMetric = allMetrics.get(metrics.metricName("remote-copy-throttle-time-max", "RemoteLogManager"));
-            assertEquals(Double.NaN, avgMetric.metricValue());
-            assertEquals(Double.NaN, maxMetric.metricValue());
+            if (quotaExceeded) {
```

Comment:
Do we need this `if (quotaExceeded) {}` block? It will always be false.

Reply:
i left it in because it looks like the original test author intended to test this logic with quotaExceeded=true as well.

Reply:
shall we remove the if..else block and replace with the below asserts? 

```
 // Metrics are not created until they actually get recorded (e.g. if the quota is exceeded).
assertNull(avgMetric);
assertNull(maxMetric);
```

because the if block is unreachable.